### PR TITLE
newdevice: Add Last Mile CTM support

### PR DESCRIPTION
**Pull request by pheinrichs**
This PR adds support for Lastmile Gear CTM. 
Please review the custom state polling I've added. 

Thanks!

~~The blocker is in place as I'm waiting to see if there is a newer MIB from the manufacturer.~~
Update: This is the newest MIB available. Custom MIB will easiest solution. For now

DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

### CODE REVIEW

File: LibreNMS/OS/Lastmile.php [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+use LibreNMS\Device\WirelessSensor;
+use LibreNMS\OS;
+
+class Lastmile extends OS implements WirelessClientsDiscovery
+{
+    /**
+     * This is a bit of a stretch. Clients are referencing `satellites`
+     * as this is a gps timing device.
+     */
+    public function discoverWirelessClients()
```
Comment (murrant) [inline LibreNMS/OS/Lastmile.php]: This seems like the complete wrong place for this data.  I'd rather not include it than put it here.

**Comment by murrant** [comment]
@pheinrichs what is this device.  It appears to be a rebranded Cambium Canopy?

**Comment by pheinrichs** [comment]
@murrant It is a GPS timing device.It is a custom firmware indeed. None of the cambium mibs / oid's would work.

**Comment by murrant** [comment]
I mean the mib looks like a copy if the canopy mib.

Ok, gps satellites does not belong in the wireless clients stat.  You can either create a custom graph or omit it

**Comment by murrant** [comment]
Also, update the os description please.

**Comment by pheinrichs** [comment]
@murrant Consider it done, 
Would you be opposed to me adding a new wireless sensor for GPS satellites?

**Comment by murrant** [comment]
@pheinrichs I think the gps satellites would be more of a generic thing.  Almost most any type of device could have gps in it.

File: includes/polling/os/lastmile.inc.php [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or (at your
+ * option) any later version.  Please see LICENSE.txt at the top level of
+ * the source code distribution for details.
+ */
+
+
+$hardware = $device['sysDescr'];
+$version = snmp_get($device, '1.3.6.1.4.1.25868.1.1.0', '-Ovqs', '');
```
Comment (laf) [inline includes/polling/os/lastmile.inc.php]: You can drop that last `''`

File: includes/definitions/lastmile.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+os: lastmile
```
Comment (laf) [inline includes/definitions/lastmile.yaml]: This looks like it's the vendors name. The OS name needs to be what the firmware is called or if that's not possible the device / type of device.

File: includes/definitions/discovery/lastmile.yaml [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+            data:
+                - 
+                    oid: tempC
+                    num_oid: .1.3.6.1.4.1.25868.1.9.
+                    index: tempC.0
+                    descr: 'Temperature'
+        power:
+            options:
+                divisor: 10
+            data:
```
Comment (laf) [inline includes/definitions/discovery/lastmile.yaml]: I'm assuming this isn't in a table? :)

Comment (pheinrichs) [reply includes/definitions/discovery/lastmile.yaml]: Boy do I wish it was...

**Pull request merged**